orderBook.js

```diff
@@ -1,5 +1,50 @@
+
 function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+  if (existingBook.length === 0) {
+    existingBook.push(incomingOrder)
+  } else {
+    var matchingIndex = -1
+    var quantityNotFulfilled = incomingOrder.quantity
+    var matchingIndexList = []
+    for (var i = 0; i < existingBook.length; i++) {
```

Comment:
Good job continuing on to the next matching order if there is still quantity left to fulfill, many didn't catch that.